modules/network/src/main/java/org/apache/ignite/network/NettyBootstrapFactory.java

```diff
@@ -139,6 +139,27 @@ public void start() {
         clientWorkerGroup = NamedNioEventLoopGroup.create(eventLoopGroupNamePrefix + "-client");
     }
 
+    /**
+     * Returns {@code true} if the current thread is a network thread, {@code false} otherwise.
+     *
+     * @return {@code true} if the current thread is a network thread, {@code false} otherwise.
+     */
+    public static boolean isInNetworkThread() {
+        String name = Thread.currentThread().getName();
+
+        if (name.contains("-srv-worker")) {
```

Comment:
It seems a bit scary to detect threads by name. It is possible to attach some thread-local flag to these threads and check using this flag?